aquarius/events/processors.py

```diff
@@ -352,3 +381,52 @@ def process(self):
         self.es_instance.update(self.asset, self.did)
 
         return self.asset
+
+
+class MetadataStateProcessor(EventProcessor):
+    def process(self):
+        self.did = make_did(self.event.address, self._chain_id)
+
+        if self.event.args.state != MetadataStates.ACTIVE:
```

Comment:
how about this?
if self.event.args.state is Active -> soft delete the asset
else: delegate to a function to put back the DDO (will improve readability)

in. the current state there is so much imbrication here I can't wrap my head around it.